wrappers/node/client-vcx/common.js

```diff
@@ -44,9 +45,40 @@ function getRandomInt (min, max) {
   return Math.floor(Math.random() * (max - min)) + min
 }
 
+async function waitUntilAgencyIsReady (agencyEndpoint, logger) {
+  let agencyReady = false
+  while (!agencyReady) {
+    try {
+      await axios.get(`${agencyEndpoint}/agency`)
+      agencyReady = true
+    } catch (e) {
+      logger.warn(`Agency ${agencyEndpoint} should return 200OK on HTTP GET ${agencyEndpoint}/agency, but returns error: ${e}. Sleeping.`)
+      await sleepPromise(1000)
+    }
+  }
+}
+
+async function pollFunction (fn, actionDescription, logger, attemptsThreshold = 10, timeout = 2000) {
+  let { result, isFinished } = await fn()
+  let attempts = 1
+  while (!isFinished) {
+    if (attempts > attemptsThreshold) {
+      const error = `Tried to poll ${attempts} times and result was not received.`
+      return [error, null]
+    }
+    logger.info(`Trying to do: ${actionDescription} Attempt ${attempts}/${attemptsThreshold}. Will try again after ${timeout}ms.`)
+    await sleepPromise(timeout);
```

Comment:
Seems like this is special case in JS where `;` is actually required. My IDE was complaining not having it there. It's probably because one line is ending with `)` next is starting with `(`